handler/oauth2/flow_refresh.go

```diff
@@ -151,8 +185,20 @@ func (c *RefreshTokenGrantHandler) PopulateTokenEndpointResponse(ctx context.Con
 		return err
 	}
 
-	if err = c.TokenRevocationStorage.CreateRefreshTokenSession(ctx, refreshSignature, storeReq); err != nil {
-		return err
+	if rtRequest, ok := requester.(fosite.RefreshTokenAccessRequester); ok {
+		rtStoreReq := requester.Sanitize([]string{}).(*fosite.Request)
+		rtStoreReq.SetID(ts.GetID())
+
+		rtStoreReq.RequestedScope = rtRequest.GetRefreshTokenRequestedScopes()
```

Comment:
Is this not already available with `ts` in this function? Do we need to perform the added tasks of populating the requester with the original requested scopes and granted scopes as rt scopes in `HandleTokenEndpointRequest`?

REF: `ts, err := c.TokenRevocationStorage.GetRefreshTokenSession(ctx, signature, nil)`

Reply:
Potentially, let me run some tests to confirm. I suspect I did this intentionally but I can't recall why and since it's not clear and I can't remember and it's important to clear that up one way or another.

Reply:
So the reason I believe did this and it actually warrants a discussion is that the difference between these two methods is that one will utilize the original Refresh Token requested at (as well as other values) value and one utilizes a new one (based on the time of the request). Realistically we're issuing a new token so at least a vast majority of the values should be based on the new request itself.

One curious question that maybe we should be asking is shouldn't the JTI be different since it's effectively a brand new token?

Reply:
Small note which is not a strong argument, but the way this is formed in the new flow is identical in effective functionality of current behavior. See here: https://github.com/ory/fosite/blob/master/handler/oauth2/flow_refresh.go#L136-L156

Reply:
Resolved, understand what you mean now.